kratos/includes/node.h

```diff
@@ -200,92 +162,76 @@ class Node : public Point, public Flags
+     * @param rOtherCoordinates The coordinates of the new node.
+     */
+    Node(const IndexType NewId, std::vector<double> const&  rOtherCoordinates);
 
+    /**
+     * @brief Constructs a 3D Node object with a given ID, X, Y, and Z coordinates, variables list, data, and queue size.
+     * @param NewId The ID of the new node.
+     * @param NewX The X coordinate of the new node.
+     * @param NewY The Y coordinate of the new node.
+     * @param NewZ The Z coordinate of the new node.
+     * @param pVariablesList The variables list of the new node.
+     * @param ThisData The data of the new node.
+     * @param NewQueueSize The queue size of the new node.
+     */
+    Node(const IndexType NewId, const double NewX, const double NewY, const double NewZ, VariablesList::Pointer  pVariablesList, BlockType const * ThisData, SizeType NewQueueSize = 1);
 
-    /** Constructor using coordinates stored in given std::vector. Initialize
-    this point with the coordinates in the array. */
-    Node(IndexType NewId, std::vector<double> const&  rOtherCoordinates)
-        : BaseType(rOtherCoordinates)
-        , Flags()
-        , mNodalData(NewId)
-        , mDofs()
-        , mData()
-        , mInitialPosition()
-        , mNodeLock()
-    {
-        CreateSolutionStepData();
-    }
-
-    /// 3d with variables list and data constructor.
-    Node(IndexType NewId, double const& NewX, double const& NewY, double const& NewZ, VariablesList::Pointer  pVariablesList, BlockType const * ThisData, SizeType NewQueueSize = 1)
-        : BaseType(NewX, NewY, NewZ)
-        , Flags()
-        , mNodalData(NewId, pVariablesList,ThisData,NewQueueSize)
-        , mDofs()
-        , mData()
-        , mInitialPosition(NewX, NewY, NewZ)
-        , mNodeLock()
-    {
-    }
-
-    typename Node::Pointer Clone()
-    {
-        Node::Pointer p_new_node = Kratos::make_intrusive<Node >( this->Id(), (*this)[0], (*this)[1], (*this)[2]);
-        p_new_node->mNodalData = this->mNodalData;
-
-        Node::DofsContainerType& my_dofs = (this)->GetDofs();
-        for (typename DofsContainerType::const_iterator it_dof = my_dofs.begin(); it_dof != my_dofs.end(); it_dof++)
-        {
-            p_new_node->pAddDof(**it_dof);
-        }
-
-        p_new_node->mData = this->mData;
-        p_new_node->mInitialPosition = this->mInitialPosition;
-
-        p_new_node->Set(Flags(*this));
-
-        return p_new_node;
-    }
+    /**
+     * @brief Clones the current Node and returns a pointer to the new Node.
+     * @return typename Node::Pointer A pointer to the new Node.
+     */
+    typename Node::Pointer Clone();
 
     /// Destructor.
-    ~Node() override
-    {
-        ClearSolutionStepsData();
-    }
+    ~Node() override;
 
     //*********************************************
     //public API of intrusive_ptr
-    unsigned int use_count() const noexcept
-    {
-        return mReferenceCounter;
-    }
+    unsigned int use_count() const noexcept;

```

Comment:
Much as I like, all these need to remain in the header. Its the same problem as with the point split. Everything that we are sure that can be inlined needs to remain in the header (Getter, Setters and non complex functions mainly).

Reply:
Done

Reply:
@loumalouomega **Everything** means ALL getters and setters (and all the functions that do simple access with no modifications)

Reply:
Okay, sorry for the noise. Anyway if we don't put inline the compiler may ignore it, specially gcc which is not very friend of inline

Reply:
Yep, we were reading it with @RiccardoRossi last week and seems that `inline`is just a hint, it may ignore it if you put it, or it may add it if you don't. Seems to be under no control. 

What is true is that if we divide it then it only can perform the inlining in the same compilation unit, so it could do it in the core (regardless of the function attributes) but won't do it under any circumstance for example in the FluidDynamics. The later is the situation we want to avoid.

Reply:
I improved the inlining